templates/flavors/system-assigned-identity/system-assigned-identity.yaml

```diff
@@ -0,0 +1,78 @@
+---
+apiVersion: cluster.x-k8s.io/v1alpha3
+kind: MachineDeployment
+metadata:
+  name: "${CLUSTER_NAME}-md-0"
+spec:
+  clusterName: "${CLUSTER_NAME}"
+  replicas: ${WORKER_MACHINE_COUNT}
+  selector:
+    matchLabels:
+  template:
+    spec:
+      clusterName: "${CLUSTER_NAME}"
+      version: "${KUBERNETES_VERSION}"
+      bootstrap:
+        configRef:
+          name: "${CLUSTER_NAME}-md-0"
+          apiVersion: bootstrap.cluster.x-k8s.io/v1alpha3
+          kind: KubeadmConfigTemplate
+      infrastructureRef:
+        name: "${CLUSTER_NAME}-md-0"
+        apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+        kind: AzureMachineTemplate
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+kind: AzureMachineTemplate
+metadata:
+  name: "${CLUSTER_NAME}-md-0"
+spec:
+  template:
+    spec:
+      location: ${AZURE_LOCATION}
+      identityType: SystemAssigned
+      vmSize: ${AZURE_NODE_MACHINE_TYPE}
+      osDisk:
+        osType: "Linux"
+        diskSizeGB: 30
+        managedDisk:
+          storageAccountType: "Premium_LRS"
+      sshPublicKey: ${AZURE_SSH_PUBLIC_KEY}
+---
+apiVersion: bootstrap.cluster.x-k8s.io/v1alpha3
+kind: KubeadmConfigTemplate
+metadata:
+  name: "${CLUSTER_NAME}-md-0"
+spec:
+  template:
+    spec:
+      joinConfiguration:
+        nodeRegistration:
+          name: '{{ ds.meta_data["local_hostname"] }}'
+          kubeletExtraArgs:
+            cloud-provider: azure
+            cloud-config: /etc/kubernetes/azure.json
+      files:
+        - path: /etc/kubernetes/azure.json
+          owner: "root:root"
+          permissions: "0644"
+          content: |
+            {
+              "cloud": "AzurePublicCloud",
+              "tenantId": "${AZURE_TENANT_ID}",
+              "subscriptionId": "${AZURE_SUBSCRIPTION_ID}",
+              "aadClientId": "${AZURE_CLIENT_ID}",
+              "aadClientSecret": "${AZURE_CLIENT_SECRET}",
```

Comment:
The azure.json file should not use Service Principal credentials, but rather, the system assigned identity (useManagedIdentity, I think).

Reply:
`useManagedIdentity` is right, see https://github.com/kubernetes-sigs/cloud-provider-azure/blob/master/docs/cloud-provider-config.md#auth-configs for reference

Reply:
yes, I was using useManagedIdentity, just didn't remove the other ones